File: abc/ABC130/cpp/d.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(){
    int N;
    long long int K;
    cin >> N >> K;
    
    vector<int> a(N);
    for(int i = 0; i < N; i++) cin >> a[i];

    long long int ans = 0;
    long long int sum = 0;

    int r = 0;
    for(int l=0; l<N; l++){
        while(sum<K){
            if(r==N)break;
            else{
                sum += a[r];
                r++;
            }
        }
        if(sum<K)break;
        ans += N-r+1;
        sum -= a[l];
    }
    cout << ans << endl;
    return 0;
}

File: abc/ABC130/python/d.py
N, K = list(map(int, input().split()))
a = list(map(int, input().split()))

sum = 0
ans = 0

r = 0
for l in range(N):
        while sum<K:
                if r == N: break
                sum += a[r]
                r += 1
        if sum<K: break
        ans += N-r+1
        sum -= a[l]
print(ans)